tests/test_integration.py

```diff
@@ -429,6 +424,117 @@ def test_multiplicative_quantile_regression_90(is_plot, prepare_data, features,
     np.testing.assert_almost_equal(quantile_acc, 0.9015, 3)
 
 
+@pytest.mark.skip(reason="Long running time")
+def test_multiplicative_quantile_regression_spline(is_plot, prepare_data, features, feature_properties):
+    X, y = prepare_data
+
+    quantiles = []
+    cdf_values = []
+    for quantile in [0.1, 0.3, 0.5, 0.7, 0.9]:
+        CB_est = cb_multiplicative_quantile_regressor_model(
+            quantile=quantile, features=features, feature_properties=feature_properties
+        )
+        CB_est.fit(X.copy(), y)
+        yhat = CB_est.predict(X.copy())
+        cdf_values.append(yhat)
+        quantiles.append(quantile)
+
+    quantiles = np.asarray(quantiles)
+    cdf_values = np.asarray(cdf_values)
+
+    i = 24
+    spl_fit_cdf = cdf_fit_spline(quantiles, cdf_values[:, i])
+
+    if is_plot:
+        plt.plot(quantiles, cdf_values[:, i], "ro")
+        xs = np.linspace(0, 1, 100)
+        plt.plot(xs, spl_fit_cdf(xs))
+        plt.savefig("spline_integration" + str(i) + ".png")
+        plt.clf()
```

Comment:
Shouldn't the test have an assert statement to guarantee that the generated CDF from the matching procedure yields the correct results? 

I might be overlooking something, but I fail to see how these tests add extra coverage beyond those defined in `test_quantile_matching.py`

Reply:
Yes, let me add some asserts :). The reason why I didn't include at first is that it is more a qualitative test here with the plots. And also a little bit of example how to use it.
The extra coverage of these tests is that we use the functionality on an actual training, i.e., quantile estimates with some uncertainty. That is a more difficult scenario than in the unit tests.